pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,477 @@
+		Expect(apierrors.IsNotFound(err)).To(BeTrue(), "Should return NotFound error")
+	})
+
+	ginkgo.It("rejects duplicate NodePool creation", func(ctx context.Context) {
+		if clusterNamespace == "" || !canAccessNodePoolAPI(h, clusterNamespace) {
+			ginkgo.Skip("NodePool API not accessible")
+		}
+
+		if !createdNodePool {
+			ginkgo.Skip("No NodePool created to test duplication")
+		}
+
+		duplicateSpec := buildNodePoolSpec(testNodePoolName, clusterNamespace)
+		duplicateObj := &unstructured.Unstructured{Object: duplicateSpec}
+
+		_, err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).
+			Create(ctx, duplicateObj, metav1.CreateOptions{})
+
+		Expect(err).To(HaveOccurred(), "Should fail creating duplicate NodePool")
+		Expect(apierrors.IsAlreadyExists(err)).To(BeTrue(), "Should return AlreadyExists error")
+	})
+})
+
+func findNodePoolNodes(ctx context.Context, client *resources.Resources) ([]corev1.Node, error) {
+	var nodeList corev1.NodeList
+	if err := client.List(ctx, &nodeList); err != nil {
+		return nil, fmt.Errorf("failed to list nodes: %w", err)
+	}
+
+	var nodePoolNodes []corev1.Node
+	for _, node := range nodeList.Items {
+		if _, hasLabel := node.Labels[nodePoolLabel]; hasLabel {
+			nodePoolNodes = append(nodePoolNodes, node)
+		}
+	}
+	return nodePoolNodes, nil
+}
+
+func findReadyNodePoolNodes(ctx context.Context, client *resources.Resources) ([]corev1.Node, error) {
+	nodes, err := findNodePoolNodes(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+
+	var readyNodes []corev1.Node
+	for _, node := range nodes {
+		if isNodeReady(node) {
+			readyNodes = append(readyNodes, node)
+		}
+	}
+	return readyNodes, nil
+}
+
+func findNodesForNodePool(ctx context.Context, client *resources.Resources, nodePoolName string) ([]corev1.Node, error) {
+	var nodeList corev1.NodeList
+	if err := client.List(ctx, &nodeList); err != nil {
+		return nil, err
+	}
+
+	var matchingNodes []corev1.Node
+	for _, node := range nodeList.Items {
+		if labelValue, exists := node.Labels[nodePoolLabel]; exists {
+			if strings.Contains(labelValue, nodePoolName) {
+				matchingNodes = append(matchingNodes, node)
+			}
+		}
+	}
+	return matchingNodes, nil
+}
+
+func validateAWSIntegration(node corev1.Node) {
+	Expect(node.Spec.ProviderID).To(HavePrefix("aws://"),
+		fmt.Sprintf("Node %s missing AWS provider ID", node.Name))
+
+	hasInternalIP := false
+	for _, addr := range node.Status.Addresses {
+		if addr.Type == corev1.NodeInternalIP && addr.Address != "" {
+			hasInternalIP = true
+			break
+		}
+	}
+	Expect(hasInternalIP).To(BeTrue(), fmt.Sprintf("Node %s missing InternalIP", node.Name))
+
+	Expect(node.Labels["node.kubernetes.io/instance-type"]).ToNot(BeEmpty(),
+		fmt.Sprintf("Node %s missing instance type", node.Name))
+}
+
+func getClusterNamespace(h *helper.H) (string, error) {
+	if name := viper.GetString(config.Cluster.Name); name != "" {
+		return "clusters-" + name, nil
```

Comment:
You're using `fmt.Sprintf` in other functions but you're not using it here.  Probably be good to be consistent.